File: src/test/sample.c
/*
nodes: 1
*/
void trivial() {}

/*
nodes: 3
*/
void simpleIf() {
    if (x) {
    }
}

/*
nodes: 6,
exits: 1
*/
void ifElse() {
    if (x) {
    } else if (y) {
    } else {
    }
}

/*
nodes: 2,
exits: 0
*/
void forever() {
    for (;;) {
    }
}

/*
nodes: 2,
exits: 0
*/
void forever2() {
    for (int a = 0;; ++a) {
    }
}

/*
nodes: 4,
exits: 1,
reaches: [["a","b"],["b","a"]]
*/
void forLoop() {
    for (int a = 0; a < 10; ++a) {
        // CFG: a
        // CFG: b
    }
}

/*
nodes: 4,
exits: 1
*/
void whileLoop() {
    while (x()) {
    }
}

/*
nodes: 3,
exits: 1
*/
void doWhile() {
    do {
    } while (x());
}

File: src/test/sample.go
package main

/*
nodes: 1
*/
func trivial() {
}

/*
nodes: 3
*/
func ifThen() {
	if x {

	}
}

/*
nodes: 9
*/
func Switch() {
	switch x {
	case 1:
	case 2:
	case 3:
	case 4:
	}
}

/*
nodes: 12
*/
func switchAndLabelsAndGoto() {
	switch x {
	case 1:
	case 2:
		goto label
	case 3:
	case 4:
	}
	return

label:
	if x {
		f()
	} else {
		g()
	}
}

/*
nodes: 1,
reaches: [
	["a", "b"]
]
*/
func trivialReachability() {
	// CFG: a
	// CFG: b
}

/*
reaches: [
	["A", "B"]
]
*/
func hasFallthrough() {
	switch 1 {
	case 1:
		// CFG: A
		fallthrough
	case 2:
		// CFG: B
		"Include me!"
	}
}